core/src/main/scala/org/ensime/config/Environment.scala

```diff
@@ -32,7 +34,5 @@ object Environment {
   private def ensimeVersion: String =
     BuildInfo.version
 
-  def shutdownOnDisconnectFlag: Boolean = {
-    Option(System.getProperty("ensime.explode.on.disconnect")).isDefined
-  }
+  def shutdownOnDisconnectFlag(implicit serverConfig: EnsimeServerConfig): Boolean = serverConfig.shutDownOnDisconnect
```

Comment:
there is no reason to have this method anymore if the caller can read it off the config object